scripts/g.extension/g.extension.py

```diff
@@ -234,6 +233,40 @@ def urlopen(url, *args, **kwargs):
     return urlrequest.urlopen(request, *args, **kwargs)
 
 
+def get_github_branches(github_api_url, version_only=False):
+    """Get ordered list of branch names in repo using github API"""
+    req = urlrequest.urlopen(github_api_url)
+    content = json.loads(req.read())
+    branches = [repo_branch["name"] for repo_branch in content]
+    if version_only:
+        branches = [
+            version_branch
+            for version_branch in branches
+            if version_branch.startswith("grass")
+        ]
+    branches.sort()
+    return branches
+
+
+def get_default_branch(full_url):
+    """Get default branch for repo (currently only implemented for github API and gitlab API"""
+    if "github.com" in full_url:
```

Comment:
Although CodeQL thinks this is a security-related context (`Incomplete URL substring sanitization`), it is right about this being not 100% robust. I'm afraid adding `://.../` is still similar which would mean using regexp. I think the split does _not_ suffer from the same issue, so maybe doing the split on first `://` but perhaps there is no reason to avoid regexp here.

Reply:
Not sure I understood correctly. But I tried to address it by using `re` and adding `/` to the host...
Hope that solves it...

Reply:
That's better, but still not perfect, although apparently CodeQL missises that. The new regexp split is doing the same as the original one since there is no regular expression there. The plain split is actually fine as is and you could use the same trick to get the domain. That's what I meant.

But there is actually a better solution:

```python
>>> from urllib.parse import urlparse  # Fine as is in Python 3
>>> urlparse("https://github.com/OSGeo/grass-addons/")
ParseResult(scheme='https', netloc='github.com', path='/OSGeo/grass-addons/', params='', query='', fragment='')
>>> url_parts = urlparse("https://github.com/OSGeo/grass-addons/")
>>> url_parts.netloc
'github.com'
>>> url_parts.netloc == "github.com"
True
```

You can parse it once and then reuse it in both tests and for both split calls.